apps/design/frontend/src/image_input.tsx

```diff
@@ -113,39 +115,131 @@ async function loadBitmapImageAndConvertToSvg(file: File): Promise<string> {
+    if (viewBox) {
+      // The value of the viewBox attribute is a list of four numbers separated by whitespace and/or a comma: min-x, min-y, width, and height.
+      const [, , viewBoxWidth, viewBoxHeight] = viewBox
+        .replace(/,/g, '')
+        .split(/\s+/);
+      width = width || viewBoxWidth;
+      height = height || viewBoxHeight;
+    }
+  }
+
+  function parseDimension(dimension: string | null) {
+    // Assume an SVG with no dimensions specified is infinitely scalable
+    if (!dimension) {
+      return Number.MAX_SAFE_INTEGER;
+    }
+
+    // Round because viewBox may specify floats as strings
+    return Math.round(
+      safeParseNumber(
+        // Remove units from string
+        dimension.replace(/[a-zA-Z]/g, '')
+      ).unsafeUnwrap()
+    );
+  }
+
+  const widthPx = parseDimension(width);
+  const heightPx = parseDimension(height);
+
+  if (minWidthPx && widthPx < minWidthPx) {
+    return err(
+      new Error(
+        `Image width (${widthPx}px) is smaller than minimum (${minWidthPx}px).`
+      )
+    );
+  }
+  if (minHeightPx && heightPx < minHeightPx) {
+    return err(
+      new Error(
+        `Image height (${heightPx}px) is smaller than minimum (${minHeightPx}px).`
+      )
+    );
+  }
+
+  return ok();
 }
 
 export function ImageInputButton({
   onChange,
+  onError,
+  minHeightPx,
+  minWidthPx,
   ...props
 }: ImageInputButtonProps): JSX.Element {
   return (
     <FileInputButton
       {...props}
       accept={ALLOWED_IMAGE_TYPES.join(',')}
       onChange={async (e) => {
-        try {
-          /* istanbul ignore next */
-          const file = assertDefined(e.target.files?.[0]);
-          if (file.size > MAX_IMAGE_UPLOAD_BYTES) {
-            throw new Error(
+        /* istanbul ignore next */
+        const file = assertDefined(e.target.files?.[0]);
+        if (file.size > MAX_IMAGE_UPLOAD_BYTES) {
+          onError(
+            new Error(
               `Image file size must be less than ${
                 MAX_IMAGE_UPLOAD_BYTES / 1_000 / 1_000
               } MB`
-            );
-          }
-          assert(ALLOWED_IMAGE_TYPES.includes(file.type));
-          const svgImage =
-            file.type === 'image/svg+xml'
-              ? await loadSvgImage(file)
-              : await loadBitmapImageAndConvertToSvg(file);
-          onChange(sanitizeSvg(svgImage));
-        } catch (error) {
-          // TODO handle errors and show to user when we do form validation
-          // eslint-disable-next-line no-console
-          console.error(error);
+            )
+          );
+          return;
         }
+        assert(ALLOWED_IMAGE_TYPES.includes(file.type));
+        const svgImage =
+          file.type === 'image/svg+xml'
+            ? await loadSvgImage(file)
+            : await loadBitmapImageAndConvertToSvg(file);
+
+        const validateResult = validateSvgDimensions(
+          svgImage,
+          minHeightPx,
+          minWidthPx
+        );
```

Comment:
I debated whether we needed to enforce image dimensions on SVGs but decided to for a few reasons

* It's simpler for the user if all image types are enforced in the same way
* I'm not sure what the limits of SVG scaling are eg. can an SVG with a defined 1x1 size scale up to 200x200? Probably, but without time to investigate properly I don't want to risk complications

Reply:
Personally, I would have stuck with just validating the size of non-SVG images. It's much easier than trying to guess whether an SVG has the size we want. For example, an SVG that says it's 1x1 pixel could possibly be scaled up to 200x200 just fine. It's also possible that a 1000x1000 pixel SVG could contain a very low-res bitmap image. Without going quite deep on it, we don't really know.

Reply:
Good points. Removed validation of SVG dimensions in

[ec27aca](https://github.com/votingworks/vxsuite/pull/5936/commits/ec27aca0b2903dee1559fd6786f28abeea6cbcde)